### ISSUE TITLE
Inconsistency in draw interaction sketch features

### ISSUE THREAD

**Issue by simonseyock** [openlayers/openlayers#15162]
**Describe the bug**
The draw sketch features are inconsistent and contain an polygon that has only 2 points sometimes.

If a polygon is drawn and the interaction is active the following features will be contained in the sketch feature source:
Before first click: 1 Point (Corresponds to point where the cursor is)
After first click: 1 Point, 1 LineString (Corresponds to the line up the position of the current cursor), 1 Polygon (This has only 2 points and does not have a correspondation that is seen on the map)
After second click: 1 Point, 1 LineString, 1 Polygon (Now this corresponds to the polygon between the two drawn points and the cursor)

**To Reproduce**
See sandbox here: https://codesandbox.io/s/draw-features-forked-3vztwp?file=/main.js

**Expected behavior**
I would expect the polygon to be only added after the second click. This way I could easily see at which step the draw process is at the moment, without counting clicks or points in the polygon. Also all geometries can be handled without the problem of polygons with to little points.

**Also**

Additionally unrelated to this behaviour it would be nice to have an event on the draw interaction like `drawstep` that is called after each handled event.

**Comment by mike-000** [comment]
The feature being drawn can also be obtained from the `drawstart` event as in https://openlayers.org/en/latest/examples/measure.html  In that example only that feature (even if it has only two points) and the cursor Point are styled as overlapping dashed lines if both the LineString and Polygon were shown together would not look good if the two sets dashes were not aligned (which would happen if you draw a counter clockwise polygon).

**Comment by simonseyock** [comment]
I see the benefit of providing the feature that will be the finished feature immediately.

I investigated a little further and found out, that the polygon only contains two points at exactly the moment the first point is drawn. It is two times the same point. Only after the cursor is moved a third coordinate is inserted.

It might be because of the drawend condition, so that if it gets hit it is not necessary to remove the cursor point.

This is a quite a minor issue. I am not really attached to this, but this was quite unexpected and also it does not feel clean that there is a polygon that will be interpreted as wrong by other libraries in between.

**Comment by simonseyock** [comment]
Maybe my main issue here is, that it is not very clear what the intended way is to work with the sketch features. There is no clear API for that.

My suggestion would be the following:
* `drawstep` event on the interaction that is called when a point is added after the drawstart (that means, condition for a mouseevent to handle is met and interaction is active)
* `sketchgeometrychange` or `geometrychange` event that passes on the change events of the geometry to the user of the interaction. This way it is not needed to keep track of the feature.
* Maybe a `getSketchFeature` method
* Make it clear in the docs that the polygons of the sketch feature might only contain two points.

So this might be more of a feature request than a bug. I can work on that if you think it's a good idea.
